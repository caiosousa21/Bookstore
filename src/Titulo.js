import React, { Component } from 'react';

class Titulo extends Component {
    render() {
        return (
            <h2 className='Titulo'>
                titulo
                <button className='Inserir'>Inserir Livro</button>
                <button className='Buscar'>Buscar Livro</button>
            </h2>
        )
    }
}
export default Titulo;