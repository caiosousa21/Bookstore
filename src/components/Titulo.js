import React, { Component } from 'react';
import FormInserir from './FormInserir'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { Creators as VisibilidadeActions } from '../store/actions/visibilidade'
import InputBuscar from './InputBuscar'

class Titulo extends Component {
    clickInserir = (e) => {
        e.preventDefault();
        this.props.mudarVisibilidadeInserir();
    }

    clickBuscar = (e) => {
        e.preventDefault();
        this.props.mudarVisibilidadeBusca();
    }

    render() {
        return (
            <div className='Titulo'>
                <h2 className='HeaderTitulo'>{this.props.title}</h2>
                <div className='InserirDiv'>
                    <button className='Inserir' onClick={this.clickInserir}>
                        Inserir Livro
                    </button>
                    {
                        this.props.inserir_visivel &&
                        <FormInserir id={this.props.id} />
                    }
                </div>
                <div className='BuscarDiv'>
                    <button className='Buscar' onClick={this.clickBuscar}>Buscar Livro</button>
                    {
                        this.props.busca_visivel &&
                        <InputBuscar />
                    }
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    inserir_visivel: state.visibilidade.inserir_visivel,
    busca_visivel: state.visibilidade.busca_visivel,
})

const mapDispatchToProps = dispatch => bindActionCreators(VisibilidadeActions, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Titulo);