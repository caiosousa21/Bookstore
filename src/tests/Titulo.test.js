import React from 'react';
import Titulo from '../components/Titulo';
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() })

describe('Titulo', () => {
    test('Renderizando Titulo', () => {
        const wrapper = shallow(<Titulo/>);
        expect(wrapper.exists()).toBe(true);
    }),
    it('Testando interface com snapshots',()=>{
        const tree = renderer.create(<Titulo/>).toJSON();
        expect(tree).toMatchSnapshot();
    }),
    it('Testando Click no botÃ£o Inserir',()=>{
        const wrapper = shallow(<Titulo />);
        let prevented = false;
        wrapper.find('button').at(0).simulate('click', {
            preventDefault: () => { prevented = true; }
        });
        expect(prevented).toBe(true);
    })
})