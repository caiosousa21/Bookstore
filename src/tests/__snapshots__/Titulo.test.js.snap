// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Titulo Testando interface com snapshots 1`] = `
<div
  className="Titulo"
>
  <h2
    className="HeaderTitulo"
  />
  <div
    className="InserirDiv"
  >
    <button
      className="Inserir"
      onClick={[Function]}
    >
      Inserir Livro
    </button>
    <form
      className="FormInserir"
      onSubmit={[Function]}
    >
      <div>
        <input
          className="InserirTitulo"
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="InserirAutor"
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="InserirEditora"
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="InserirData"
          onChange={[Function]}
          type="Date"
          value=""
        />
      </div>
      <input
        type="submit"
      />
    </form>
  </div>
  <button
    className="Buscar"
  >
    Buscar Livro
  </button>
</div>
`;
